packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts

```diff
@@ -119,6 +129,57 @@ export function createTimeSeries(
   };
 }
 
+function transformResource(
+  resource: Resource,
+  projectId: string
+): MonitoredResource {
+  const templateResource = getTypeAndMappings(resource);
+  const type = templateResource.type;
+  const labels: { [key: string]: string } = { project_id: projectId };
+
+  for (const key of Object.keys(templateResource.labels)) {
+    if (!resource.labels[templateResource.labels[key]]) {
+      return { type: 'global', labels: { project_id: projectId } };
+    } else {
+      if (
+        type === AWS_EC2_INSTANCE &&
+        templateResource.labels[key] === CLOUD_RESOURCE.REGION
+      ) {
+        labels[key] = `${AWS_REGION_VALUE_PREFIX}${
+          resource.labels[templateResource.labels[key]]
+        }`;
+      } else {
+        labels[key] = `${resource.labels[templateResource.labels[key]]}`;
+      }
+    }
+  }
+  return { type, labels };
+}
+
+function getTypeAndMappings(resource: Resource): MonitoredResource {
+  const cloudProvider = `${resource.labels[CLOUD_RESOURCE.PROVIDER]}`;
+  // These are the only supported resources
```

Comment:
Optional: Consider specifying further what you mean by "These". Further, I think that this comment may work better as a function comment; that way, at first glance the caller knows which resources are supported by the function.